public/app/features/library-panels/components/PanelLibraryOptionsGroup/PanelLibraryOptionsGroup.tsx

```diff
@@ -28,8 +28,10 @@ export const PanelLibraryOptionsGroup: React.FC<Props> = ({ panel, dashboard })
       libraryPanel: toPanelModelLibraryPanel(panelInfo),
     });
 
-    // dummy change for re-render
-    // onPanelConfigChange('isEditing', true);
+    // Though the panel model has changed, since we're switching to an existing
+    // library panel, we reset the "hasChanged" state.
+    panel.hasChanged = false;
```

Comment:
this looks like the only place we set it to false, won't we need to set this to false when we go into panel edit mode as well to prevent the same unsaved change warning to show again?

Reply:
Yep, thanks for catching this. Should be fixed now.